### ISSUE TITLE
Make grype-db download smaller by switching compression methods

### ISSUE THREAD

**Issue by willmurphyscode** [anchore/grype-db#367]
**What would you like to be added**:

Grype should download a smaller file during it's database update, probably by using .zstd compression on the current database schema.

**Why is this needed**:

The Grype database has grown over the years, to the point where now the database is 184 mb as a gzipped tar. This results in load on the CDN, and poor experience for many users.

**Tasks**:

- [x] update yardstick to be able to handle zstd files https://github.com/anchore/yardstick/pull/441
- [x] ~update grype-db-manager config shape to include a suffix per schema version, and pass that suffix as `-e` to `grype-db package`~
- [x] ~update that configuration to pass `-e tar.zstd` for v4 and v5 schemas~ code responds to this, no config update necessary

**Comment by wagoodman** [comment]
We may be able to use Xz in a performant way instead if we use https://github.com/xi2/xz

**Comment by wagoodman** [comment]
Prototype for grype is here https://github.com/anchore/grype/compare/main...fast-xz

What's missing is removing some of the copied `untar` code from go-getter and leveraging the stereoscope tar utils (may require some refactoring in stereoscope).

**Comment by jonjohnsonjr** [comment]
I would highly recommend using zstandard over xz!

**Comment by wagoodman** [comment]
Me too! When evaluating I've been trying to minimize file size while not impacting decompression time in grype. Something that threw a wrench into this evaluation process is when to use golang implementations for these methods vs shelling out to tooling to do this. I've found when compressing with a golang implementation there tends to be less ideal compression ratios and decompression times.  The lesson here learned is: compress with native tooling (for best archives), decompress with golang implementations (allowing us to keep grype as a portable static binary easily). I also found that what is being compressed is pretty sensitive to the compression ratio too, so while we've been prototyping a new schema, we ended up changing a lot of the details based on apparent ratios we were getting with those designs (for instance, a more normalized DB design tended to be a smaller DB file size, but not a great compression ratio when making the archive for distribution... but relaxing normalization and leaning more towards a json blob store the ratio was maximized).

So! Where are we at today with all of the feedback incorporated? In terms of distribution sizes:
```
original:
711M    build/vulnerability-db_v6.0.0_2024-11-14T01:32:00Z_1732070774.tar

archives:
 81M    build/vulnerability-db_v6.0.0_2024-11-14T01:32:00Z_1732070774-19.tar.zst    # zstd -19
 63M    build/vulnerability-db_v6.0.0_2024-11-14T01:32:00Z_1732070774-22.tar.zst    # zstd -22 --ultra
 58M    build/vulnerability-db_v6.0.0_2024-11-14T01:32:00Z_1732070774.tar.xz        # xz -9
```

Where Xz and Zstd-22 are comparable enough to be candidates here.


And timing (after trying out / swapping some decompression libs... I'll spare folks the details here):
```
❯ time grype db import vulnerability-db_v6.0.0_2024-11-14T01:32:00Z_1732070774.tar.xz
grype db import   6.32s user 0.34s system 89% cpu 7.488 total

❯ time grype db import vulnerability-db_v6.0.0_2024-11-14T01:32:00Z_1732070774-19.tar.zst
grype db import   1.21s user 0.58s system 168% cpu 1.065 total

❯ time grype db import vulnerability-db_v6.0.0_2024-11-14T01:32:00Z_1732070774-22.tar.zst
grype db import   2.89s user 0.46s system 130% cpu 2.574 total
```

From a timing perspective Zstd wins here.

Overall, the final verdict is Zstd 🎉
